File: 8.cs
class isPrime
{
    public bool is_prime(int n)
    {
        if (n == 1) return false;
        for (int i = 2; i < n; i++)
        {
            if (n % i == 0) return false;
        }
        return true;
    }
    public static int Main()
    {
        string input, trimmed_input, q = "q", Q = "Q";
        int limit = 0;
        const int fieldWidthLeftAligned = -9;
        bool is_parsable, flag = false;
        ConsoleColor foreground = Console.ForegroundColor;
        Console.ForegroundColor = ConsoleColor.Green;
        Console.WriteLine("@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@\n");
        Console.WriteLine("\t\t\t\t\t\tPRIME NUMBERS\n");
        Console.WriteLine("@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@\n");
        do
        {
            Console.Write("\n\nEnter the limit (or enter 'q' to exit): ");
            input = Console.ReadLine();
            trimmed_input = String.Concat(input.Where(c => !Char.IsWhiteSpace(c)));
            if (input.Equals(q) || input.Equals(Q)) System.Environment.Exit(0);
            is_parsable = Int32.TryParse(input, out limit);
            if (!is_parsable)
            {
                Console.WriteLine("\nInvalid input.");
                flag = true;
            }
            else
            {
                flag = false;
                var ip = new isPrime();
                //print prime numbers
                Console.WriteLine("\n");
                for (int k = 0; k <= limit; k++)
                {
                    if (ip.is_prime(k)) Console.WriteLine($"{k, fieldWidthLeftAligned}");
                }
            }
        } while (flag == true);
        return 0;
    }
}

File: 8.c
#include <stdio.h>
#include <stdbool.h>

// Funtion removing spaces from string
char* removeSpacesFromStr(char* string)
{
	// non_space_count to keep the frequency of non space characters
	int non_space_count = 0;

	//Traverse a string and if it is non space character then, place it at index non_space_count
	for (int i = 0; string[i] != '\0'; i++)
	{
		if (string[i] != ' ')
		{
			string[non_space_count] = string[i];
			non_space_count++;//non_space_count incremented
		}
	}

	//Finally placing final character at the string end
	string[non_space_count] = '\0';
	return string;
}
bool isPrime(int n) {
	if (n == 1) return false;
	for (int k = 2; k < n; k++) {
		if (n % k == 0) return false;
	}
	return true;
}
int main() {

	char limit[100];
	bool flag;
	int var_limit, non_space_cnt = 0;

	printf("@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@\n\n");
	printf("\t\t\t\t\t\tPRIME NUMBERS\n\n");
	printf("@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@\n\n");
	
	do {
		printf("\n\nEnter the limit (or enter 'q' to exit): ");
		fgets(limit, 100, stdin);
		removeSpacesFromStr(limit);
		if (limit[0] == 'q' || limit[0] == 'Q') {
			printf("\nChao :^)\n");
			exit(0);
		}
		var_limit = atoi(limit);
		if (var_limit < 1) {
			printf("Not valid input. Try again.");
			flag = true;
		}
		else {
			flag = false;
		}
	} while (flag == true);

	printf("\n");

	for (int j = 1; j <= var_limit; j++) {
		if (isPrime(j)) printf("%-9d ", j);
	}
	printf("\n\n");

	return 0;
}
